### PR TITLE
Jetpack Manage: Adding a new sites submenu to the Jetpack Cloud sidebar navigation

### PR DESCRIPTION
**Pull request by coder-karen**
Related to https://github.com/Automattic/jetpack-manage/issues/286

## Proposed Changes

* This PR adds a new sites submenu to the Jetpack Cloud sidebar navigation.
* The submenu items go to the following pages:
  * Needs attention => sites?issue_types=all_issues
  * Favorites => /sites/favorites
  * All => /sites
 * The default sub-menu item is now 'all'. This can be changed to 'needs attention' for example, if needed.
* Challenges in implementation included the 'is-active' menu item behaviour - because all sub-menu items are technically part of the same page all would be highlighted with the previous is-active behaviour, but the implementation turned out to be a one-line change in sites.tsx (changing the default isSelected behaviour there).
* This PR does not include an icon with the number of sites needing attention as that can be followed up with later when an API exists to get the information (it would involve a workaround otherwise). See p1707910367775269/1707846443.511099-slack-CTXBP902X for more context on that decision.

Relevant links:
PT: pfunGA-4t-p2

## Testing Instructions

* To test, apply this PR and navigate to `http://jetpack.cloud.localhost:3000/sites`. Alternatively, when the parent PR is merged, you may be able to use the banner to select the new V2 sites dashboard.
* Click on the 'Sites' menu item, and see the sidebar visible. The default sub menu item is currently 'All'. On clicking each menu item, that should then show as the new active menu item.
* The menu items have the behaviour covered under 'Proposed Changes' above.
* Clicking the 'back' chevron should take you to the Overview page.

#### Screenshots

The main sidebar (currently showing Sites V2):

Screenshot 2024-02-15 at 16 17 47

The sites sidebar:

Screenshot 2024-02-15 at 16 18 33

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by cleacos** [comment]
Testing it now...

**Comment by cleacos** [comment]
Clicking on the submenus works well. The filter is selected and shows the expected sites.

> Q - I have the default sub-menu item as 'Needs Attention'. Should this be something else, such as 'All'?

I'm not sure about the answer, but I expect to see all sites first and then switch between the different filters. But we have the problem that currently, the `/dashboard` is the main page in Manage, so for a better UX the Overview should be the main one (the new Overview page).

**Summary:**

- So, I expect to see the Sites submenu if I select `All`
- If I go to the `/dashboard` URL directly, I expect to see the submenu.
- If I apply a filter different to `all_issues`, it shows the main menu, instead of remaining on the submenu.

I suggest not merging this PR for now. Wait until the Prototype PR is merged into the trunk and put the submenu behind the feature flag. We'll probably need some feedback from designers.

File: client/jetpack-cloud/sections/agency-dashboard/controller.tsx
Comment (cleacos) [inline client/jetpack-cloud/sections/agency-dashboard/controller.tsx]: Maybe we could change this code by this. I think it should work:

```
	context.header = <Header />;
	context.secondary = showSitesDashboardV2 ? (
		<SitesSidebar path={ context.path } />
	) : (
		<JetpackManageSidebar path={ context.path } />
	);
```

Comment (coder-karen) [reply client/jetpack-cloud/sections/agency-dashboard/controller.tsx]: Works well, thanks! Added in b52212a0192a6a6c66655ac6bb82ecf54a9c4fb0

**Comment by coder-karen** [comment]
This PR has now been updated to be based off the prototype, making use of the new sites route.

**Review by cleacos — APPROVED** [review]
It works perfectly, as it is expected.

Also I tested other filters like: `/sites?issue_types=plugin_updates` and it worked as expected.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11359912

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @coder-karen for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11359912).

**Pull request merged**
